app/Listeners/readRecent.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Listeners;
+
+use App\Models\Recent;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Support\Facades\Redirect;
+use Illuminate\Contracts\Queue\ShouldQueue;
+
+class readRecent
+{
+
+    /**
+     * Handle the event.
+     */
+    public function handle(object $event)
+    {
+        
+        $book_id = $event->book_id;
+        $chapter_no = $event->chapter_no;
+        $user_id = Auth::user()->user_ID;
+        $recent = Recent::where('book_id', $book_id)
+            ->where('user_id', $user_id)
+            ->first();
+        
+        if ($recent === null) {
```

Comment:
incearca sa folosesti !$recent instanceof Recent in loc. Asa te asiguri ca $recent e obiect cu un anumit type. fi cat mai explicit unde poti